controllers/configurationpolicy_controller.go

```diff
@@ -579,10 +567,31 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 			}
 		}
 
+		if isNamespaced && len(relevantNamespaces) == 0 {
+			// If relevantNamespaces is empty, retrieve the namespaces based on filters in NamespaceSelector
+			selector := plc.Spec.NamespaceSelector
+			// If MatchLabels/MatchExpressions/Include were not provided, return no namespaces
+			if selector.MatchLabels == nil && selector.MatchExpressions == nil && len(selector.Include) == 0 {
+				log.Info("NamespaceSelector is empty. Skipping namespace retrieval.")
+			} else {
+				relevantNamespaces, err = getPolicyNamespaces(selector)
+				if err != nil {
+					log.Error(err, "Error filtering namespaces with provided NamespaceSelector", "index", indx)
```

Comment:
It'd be nice to log the invalid namespace selector as well.
```suggestion
					log.Error(err, "Error filtering namespaces with provided namespaceSelector", "index", indx, "namespaceSelector": fmt.Sprintf("#v", selector))
```

Reply:
Made the change now, but making sure I'm not crazy. This should use a comma and `%`, correct? Like so?
```
"namespaceSelector", fmt.Sprintf("%#v", selector)
```

Reply:
Sorry, that was a typo. I mean `%v`.